packages/react-components/react-shared-contexts/src/CustomStyleHooksContext/CustomStyleHooksContext.ts

```diff
@@ -0,0 +1,184 @@
+  useMenuPopoverStyles_unstable: (state: unknown) => void;
+  useMenuSplitGroupStyles_unstable: (state: unknown) => void;
+  usePersonaStyles_unstable: (state: unknown) => void;
+  usePopoverSurfaceStyles_unstable: (state: unknown) => void;
+  useRadioGroupStyles_unstable: (state: unknown) => void;
+  useRadioStyles_unstable: (state: unknown) => void;
+  useSelectStyles_unstable: (state: unknown) => void;
+  useSliderStyles_unstable: (state: unknown) => void;
+  useSpinButtonStyles_unstable: (state: unknown) => void;
+  useSpinnerStyles_unstable: (state: unknown) => void;
+  useSwitchStyles_unstable: (state: unknown) => void;
+  useTabStyles_unstable: (state: unknown) => void;
+  useTabListStyles_unstable: (state: unknown) => void;
+  useTextStyles_unstable: (state: unknown) => void;
+  useTextareaStyles_unstable: (state: unknown) => void;
+  useTooltipStyles_unstable: (state: unknown) => void;
+  useDialogTitleStyles_unstable: (state: unknown) => void;
+  useDialogBodyStyles_unstable: (state: unknown) => void;
+  useDialogActionsStyles_unstable: (state: unknown) => void;
+  useDialogSurfaceStyles_unstable: (state: unknown) => void;
+  useDialogContentStyles_unstable: (state: unknown) => void;
+  useProgressBarStyles_unstable: (state: unknown) => void;
+  useToolbarButtonStyles_unstable: (state: unknown) => void;
+  useToolbarRadioButtonStyles_unstable: (state: unknown) => void;
+  useToolbarGroupStyles_unstable: (state: unknown) => void;
+  useToolbarToggleButtonStyles_unstable: (state: unknown) => void;
+  useToolbarDividerStyles_unstable: (state: unknown) => void;
+  useToolbarStyles_unstable: (state: unknown) => void;
+  useTableCellStyles_unstable: (state: unknown) => void;
+  useTableRowStyles_unstable: (state: unknown) => void;
+  useTableBodyStyles_unstable: (state: unknown) => void;
+  useTableStyles_unstable: (state: unknown) => void;
+  useTableHeaderStyles_unstable: (state: unknown) => void;
+  useTableHeaderCellStyles_unstable: (state: unknown) => void;
+  useTableResizeHandleStyles_unstable: (state: unknown) => void;
+  useTableSelectionCellStyles_unstable: (state: unknown) => void;
+  useTableCellActionsStyles_unstable: (state: unknown) => void;
+  useTableCellLayoutStyles_unstable: (state: unknown) => void;
+  useDataGridCellStyles_unstable: (state: unknown) => void;
+  useDataGridRowStyles_unstable: (state: unknown) => void;
+  useDataGridBodyStyles_unstable: (state: unknown) => void;
+  useDataGridStyles_unstable: (state: unknown) => void;
+  useDataGridHeaderStyles_unstable: (state: unknown) => void;
+  useDataGridHeaderCellStyles_unstable: (state: unknown) => void;
+  useDataGridSelectionCellStyles_unstable: (state: unknown) => void;
+};
+
+/**
+ * @internal
+ */
+export const CustomStyleHooksContext = React.createContext<CustomStyleHooksContextValue | undefined>(undefined);
+
+const customStyleHooksContextDefaultValue: CustomStyleHooksContextValue = {
+  useAccordionHeaderStyles_unstable: () => {},
```

Comment:
Do these hooks each need a separate arrow function or can we use a single `noop` function? E.g.,

```ts
const noop = () => {};
useAccordionHeaderStyles_unstable: noop,
// and so on and so forth :)
```

Definitely on the micro-optimization side of things but it will save us a few bytes in bundle size and some bytes in memory as well.